### PR TITLE
Assignment1

### PR DESCRIPTION
**Pull request by Cooperwutzke**
Assignment 1

No extra credit attempted.

### CODE REVIEW

**Comment by StevenZuelke** [comment]
I am reviewing

File: SecretSanta/tests/SecretSanta.Business.Tests/UserTester.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        {
+            //Arrange
+            int id = 123;
+            string firstName = "RandomFirst";
+            string lastName = null;
+            List<Gift> gifts = new List<Gift>();
+
+            //Act
+
+            User testUser = new User(id, firstName, lastName, gifts);
```
Comment (StevenZuelke) [inline SecretSanta/tests/SecretSanta.Business.Tests/UserTester.cs]: Good to test these null checks, doing every part of the class.

File: SecretSanta/src/SecretSanta.Web/Startup.cs [hunk 2]
```diff
@@ -27,7 +27,7 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
             {
                 endpoints.MapGet("/", async context =>
                 {
-                    await context.Response.WriteAsync("Hello from Web!");
+                    await context.Response.WriteAsync("Hello from Web!").ConfigureAwait(false);
```
Comment (StevenZuelke) [inline SecretSanta/src/SecretSanta.Web/Startup.cs]: could have suppressed the warning on this one

File: SecretSanta/src/SecretSanta.Api/SecretSanta.Api.csproj [hunk 3]
```diff
@@ -4,6 +4,18 @@
     <TargetFramework>netcoreapp3.1</TargetFramework>
   </PropertyGroup>
```
Comment (StevenZuelke) [inline SecretSanta/src/SecretSanta.Api/SecretSanta.Api.csproj]: You want to configure nullability based on not already being set here.

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    restoreSolution: '$(solution)'
+
+- task: VSBuild@1
+  inputs:
+    solution: '$(solution)'
+    msbuildArgs: '/p:DeployOnBuild=true /p:WebPublishMethod=Package /p:PackageAsSingleFile=true /p:SkipInvalidConfigurations=true /p:PackageLocation="$(build.artifactStagingDirectory)"'
+    platform: '$(buildPlatform)'
+    configuration: '$(buildConfiguration)'
+
+- task: VSTest@2
```
Comment (Keboo) [inline azure-pipelines.yml]: Because you have the `DotNetCoreCLI@2` that will run the unit tests you don't need this one here.

Comment (Cooperwutzke) [reply azure-pipelines.yml]: https://dev.azure.com/cwutzke/CSCD379-Winter-2020/_build?definitionId=1&_a=summary

Hey Kevin, hopefully you find this. Here is the link for my Azure DevOps.

**Pull request closed without merging**